clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/userdata/LocalFileUserDataProvider.groovy

```diff
@@ -24,38 +26,80 @@ class LocalFileUserDataProvider implements UserDataProvider {
   @Autowired
   LocalFileUserDataProperties localFileUserDataProperties
 
+  @Autowired
+  Front50Service front50Service
+
+  boolean isLegacyUdf(String account, String applicationName) {
+    Closure<Boolean> result = {
+      List<Map> creds = front50Service.getCredentials()
+      Map cred = creds.find { it.global } ?: creds.find { it.name == account }
+      if (!cred) {
+        throw new IllegalStateException("Unable to find front50 credentials (global of $account)")
+      }
+      String front50Account = cred.name
+
+      try {
+        Map application = front50Service.getApplication(front50Account, applicationName)
+        if (application.legacyUdf == null) {
+          return localFileUserDataProperties.defaultLegacyUdf
+        }
+        return Boolean.valueOf(application.legacyUdf)
+      } catch (RetrofitError re) {
+        if (re.kind == RetrofitError.Kind.HTTP && re.response.status == 404) {
+          return localFileUserDataProperties.defaultLegacyUdf
+        }
+        throw re
+      }
+    }
+
+    final int maxRetry = 5
+    final int retryBackoff = 500
+    final Set<Integer> retryStatus = [429, 500]
+    for (int i = 0; i < maxRetry; i++) {
+      try {
+        return result.call()
+      } catch (RetrofitError re) {
+        if (re.kind == RetrofitError.Kind.NETWORK || (re.kind == RetrofitError.Kind.HTTP && retryStatus.contains(re.response.status))) {
+          Thread.sleep(retryBackoff)
+        }
+      }
+    }
+    throw new IllegalStateException("Failed to read legacyUdf preference from front50 for $account/$applicationName")
+  }
+
   @Override
   String getUserData(String asgName, String launchConfigName, String region, String account, String environment, String accountType) {
     def names = Names.parseName(asgName)
-    def rawUserData = assembleUserData(names, region, account)
-    replaceUserDataTokens localFileUserDataProperties.useAccountNameAsEnvironment, names, launchConfigName, region, account, environment, accountType, rawUserData
+    boolean legacyUdf = isLegacyUdf(account, names.app)
+    def rawUserData = assembleUserData(legacyUdf, names, region, account)
+    replaceUserDataTokens legacyUdf, names, launchConfigName, region, account, environment, accountType, rawUserData
   }
 
-  String assembleUserData(Names names, String region, String environment) {
-    def udfRoot = localFileUserDataProperties.udfRoot
+  String assembleUserData(boolean legacyUdf, Names names, String region, String account) {
```

Comment:
renamed environment -> account to be clear that is what we were templating on (not a change in behaviour)